### ISSUE TITLE
[java] SimplifyBooleanReturns should check duplicated function call in if statement

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2581]
**Affects PMD Version:** 
6.22.0

**Rule:**
SimplifyBooleanReturns

**Description:**
It is recommended to have SimplifyBooleanReturns additionally check the case like <code>if(condition){return function(true);}else{return function(false);}</code>
Similar to the rule detection content,
```
if(condition){
	return function(true);
}
else{
	return function(false);
}
```
The above situation can also be simplified to <code>return function(condition);</code>

**Code Sample demonstrating the suggestion:**
Example 1:
```
// can also be simplified to "return setGodMode(player, isGod(player))"
if (isGod(player)) {
	return setGodMode(player, false);
} else{
	return setGodMode(player, true);
}
```

Example 2:
```
// can also be simplified to "getIntersection(ed.startPoint, ed.endPoint, wo>wt)))"
if(wo>wt){
	return getIntersection(ed.startPoint, ed.endPoint,true);
} else {
	return getIntersection(ed.startPoint, ed.endPoint, false);
}
```

**Type:**
Suggestions for rule improvement

**Running PMD through:**
CLI

**Comment by oowekyala** [comment]
Not sure this really belongs in SimplifyBooleanReturns... By the same logic

```java
if (<condition>) {
  doSomething(<boolean>);
} else{
  doSomething(<boolean>);
}
```

without returns could be flagged

**Comment by oowekyala** [comment]
Closing this in favour of #2582. A priori both these cases would be solved in the same way
